frontend/lib/src/components/shared/Dropdown/Selectbox.test.tsx

```diff
@@ -39,152 +39,121 @@ const getProps = (props: Partial<Props> = {}): Props => ({
+    expect(screen.getByTestId("stWidgetLabel")).toHaveStyle("display: none")
   })
 
   it("pass placeholder prop correctly", () => {
     props = getProps({
+      value: undefined,
       placeholder: "Please select",
     })
-    wrapper = shallow(<Selectbox {...props} />)
-
-    expect(wrapper.find(UISelect).prop("placeholder")).toEqual("Please select")
+    render(<Selectbox {...props} />)
+    expect(screen.getByText("Please select")).toBeInTheDocument()
   })
 
   it("renders a placeholder with empty options", () => {
     props = getProps({
       options: [],
     })
-    wrapper = shallow(<Selectbox {...props} />)
+    render(<Selectbox {...props} />)
 
-    expect(wrapper.find(UISelect).prop("options")).toStrictEqual([
-      {
-        label: "No options to select.",
-        value: "0",
-      },
-    ])
+    expect(screen.getByText("No options to select.")).toBeInTheDocument()
+    expect(screen.getByRole("combobox")).toBeDisabled()
   })
 
   it("renders options", () => {
-    const options = wrapper.find(UISelect).prop("options") || []
-
-    // @ts-expect-error
-    options.forEach(option => {
-      expect(option).toHaveProperty("label")
-      expect(option).toHaveProperty("value")
+    render(<Selectbox {...props} />)
+    const selectbox = screen.getByRole("combobox")
+    fireEvent.click(selectbox)
+    const options = screen.getAllByRole("option")
+
+    expect(options).toHaveLength(props.options.length)
+    options.forEach((option, index) => {
+      expect(option).toHaveTextContent(props.options[index])
     })
-
-    expect(options.length).toBe(props.options.length)
-    expect(wrapper.find(UISelect).prop("labelKey")).toBe("label")
-    expect(wrapper.find(UISelect).prop("valueKey")).toBe("value")
   })
 
   it("could be disabled", () => {
-    expect(wrapper.find(UISelect).prop("disabled")).toBe(props.disabled)
-  })
-
-  it("is able to select an option", () => {
-    // @ts-expect-error
-    wrapper.find(UISelect).prop("onChange")({
-      value: [{ label: "b", value: "1" }],
-      option: { label: "b", value: "1" },
-      type: "select",
-    })
-
-    expect(wrapper.find(UISelect).prop("value")).toContainEqual({
-      label: "b",
-      value: "1",
+    props = getProps({
+      disabled: true,
     })
+    render(<Selectbox {...props} />)
+    expect(screen.getByRole("combobox")).toBeDisabled()
   })
 
-  it("doesn't lose the selected value when the input is empty", () => {
-    const wrapper = mount(<Selectbox {...props} />)
-
-    const input = wrapper.find("input")
-
-    input.simulate("change", { target: { value: "" } })
-    expect(wrapper.state("isEmpty")).toBe(true)
-    expect(wrapper.state("value")).toEqual(0)
+  it("is able to select an option", () => {
+    render(<Selectbox {...props} />)
+    const selectbox = screen.getByRole("combobox")
+    // Open the dropdown
+    fireEvent.click(selectbox)
+    const options = screen.getAllByRole("option")
+    // Select the second option
```

Comment:
Super nit: i don't think this comment is necessary as it just repeats what the code says. I think we should delete it because of this principle:
https://en.wikipedia.org/wiki/Don%27t_repeat_yourself